src/index.ts

```diff
@@ -153,29 +153,67 @@ const getAssetFromKV = async (event: FetchEvent, options?: Partial<Options>): Pr
 
   if (response) {
     let headers = new Headers(response.headers)
-    headers.set('CF-Cache-Status', 'HIT')
     if (shouldSetBrowserCache) {
       headers.set('cache-control', `max-age=${options.cacheControl.browserTTL}`)
     } else {
       // don't assume we want same cache behavior of edge TTL on client
       // so remove the header from the response we'll return
       headers.delete('cache-control')
     }
-    response = new Response(response.body, { headers })
+
+    let shouldRevalidate = false
+    // Four preconditions must be met for a 304 Not Modified:
+    // - the request cannot be a range request
+    // - client sends if-none-match
```

Comment:
`If-Modified-Since` requires that the origin apply a `Last-Modified` response header. We are not applying `Last-Modified`, so `If-Modified-Since` won't ever be sent by the client. Using `if-modified-since` and `if-none-match` is redundant and `if-none-match` takes priority if both headers are sent in the request.